### ISSUE TITLE
`dim,StoredSpatialImage-method` should use `magick::image_info(img)` if raster is not loaded

### ISSUE THREAD

**Issue by LTLA** [drighelli/SpatialExperiment#112]
This primarily stems from `StoredSpatialImage`s having a very slow `dim` method (and thus `show` method) if the image has not already been loaded into the cache, because it currently needs to obtain a raster to query the `dim`.

This is most noticeable if you're trying to inspect the `imgData` and it contains multiple `StoredSpatialImage`s objects, at which point all of them are inadvertently loaded into memory as part of the DataFrame's `show` method. This defeats the purpose of having `StoredSpatialImage`s, which is to only load specific images into memory on an as-needed basis.

I believe it should be possible to specialize the `dim,StoredSpatialImage-method` to do something like:

- Call `.get_from_cache(..., NULL)` to check whether the image has already been loaded into the cache.
- If yes (i.e., the return value is not `NULL`), get the dimensions from it directly.
- Otherwise, call `magick::image_read()` and then `magick::image_info()` to quickly extract the width and height.

**Comment by HelenaLC** [comment]
Huh, smart. This? ...Now sure the unit test is the best (basically just testing that `dim()` is nearly instant on the SPI, which was not the case before actually, even with the tiny example image. 
```
#' @export
#' @importFrom magick image_read image_info
setMethod("dim",
    "StoredSpatialImage",
    function(x) {
        src <- normalizePath(imgPath(x))
        src <- paste0("file://", src)
        img <- .get_from_cache(src, NULL)
        if (!is.null(img)) return(dim(img))
        img <- image_read(src)
        tib <- image_info(img)
        c(tib$height, tib$width)
    })
```

**Comment by LTLA** [comment]
That looks right. If you want to test it, I would do the following:

- Flush the cache with `SpatialExperiment:::.flush_cache`.
- Run `dim()` for an StoredSpatialImage and check that this is correct.
- Check that the cache has not been filled with the corresponding ID.

**Comment by HelenaLC** [comment]
Should be addressed by #113; thanks (!!) Aaron for noting & the very clear instructions for resolving this.
